Dockerfile

```diff
@@ -62,3 +61,6 @@ RUN export REAL_CXX=g++-10 && export CC=gcc-10 && \
 RUN echo 'alias joe="jupp --wordwrap"' >> ~/.bashrc
 RUN echo 'export PS1="[afl++]$PS1"' >> ~/.bashrc
 ENV IS_DOCKER="1"
+
+COPY --from=aflplusplus/afl-dyninst /usr/local/lib/libdyninstAPI_RT.so /usr/local/lib/libdyninstAPI_RT.so
+COPY --from=aflplusplus/afl-dyninst /afl-dyninst/libAflDyninst.so /usr/local/lib/libAflDyninst.so
```

Comment:
Does this actually work? Wont this pull another container? I am not a docker expert ...

Reply:
See [here](https://docs.docker.com/engine/reference/builder/#copy) for how `--from` works. During the build it indeed pulls that other image, but after the 2 files are added it is no longer required and only the 2 files become part of the build.
So anyone pulling `aflplusplus/aflplusplus` won't have to pull, in this case, `aflplusplus/afl-dyninst`.

Reply:
OK. in this case please add copying afl-dyninst plus the two .sh helper scripts, otherwise people cannot use it.

Reply:
You propose to add afl-dyninst entirely into the container? Adding the two `.so` files should be enough to run any binaries instrumented by afl-dyninst. In that way the two images clearly have a different purpose, and afl++ never needs afl-dyninst during runtime as far as I can tell, right?

Reply:
my thinking is that people then do not need an own afl-dyninst container anymore, but that thinking was flawed as lots of other dyninst libraries are then required as well.

Reply:
Awesome! Still, to have this succeed we need the `aflplusplus/afl-dyninst` image to be built from https://github.com/vanhauser-thc/afl-dyninst

I'd add it as an issue there, but issues are disabled in that repo.